soc/board_specific_workflows/ice40up5k.py

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python3
+# Copyright 2021 The CFU-Playground Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import general
+import warnings
+from litex.build import generic_programmer
+from litex.soc.integration import soc as litex_soc
+from litex.soc.integration import builder
+from typing import Callable
+
+
+class Ice40UP5KWorkflow(general.GeneralSoCWorkflow):
+    """Workflow for boards derived from the Ice40UP5K.
+    
+    This includes the iCEbreaker, Fomu, TinyFPGA, and a few more. All of these
+    derived boards require a bios_flash_offset argument. They also cannot fit
+    entire CFU-Playground into RAM, so they must put read-only data in flash.
+    """
+    def __init__(self,
+                 args: argparse.Namespace,
+                 soc_constructor: Callable[..., litex_soc.LiteXSoC],
+                 builder_constructor: Callable[..., builder.Builder] = None,
+                 warn: bool = True) -> None:
+
+        # Currently only fomu variant is supported.
+        if warn and args.cpu_variant != 'fomu+cfu':
+            warnings.warn('Only fomu+cfu variant of Vexriscv supported for' +
+                          f'{type(self).__name__}, not {args.cpu_variant}.\n' +
+                          'Using fomu+cfu.')
+            args.cpu_variant = 'fomu+cfu'
```

Comment:
I think this should be outside the if statement so that it always happens? Otherwise the value of the "warn" parameter is changing the code behavior, not just warning about it.

Reply:
Good catch, fix pushed.